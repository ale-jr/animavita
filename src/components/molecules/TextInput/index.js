import propTypes from "prop-types";
import { useRef } from "react";
import {
  TextInput as BaseTextInput,
  TextInputBox,
  TextInputLabel
} from "./style";

function TextInput({ value, onChange, label, id, placeholder  }) {
  const textInputId = useRef(id ?? `search-input-${Date.now()}`);
  return (
    <TextInputBox >
      <TextInputLabel htmlFor={textInputId.current}>
        {label}
      </TextInputLabel>
      <BaseTextInput
        placeholder={placeholder}
        id={textInputId.current}
        value={value}
        onChange={onChange}
      />
    </TextInputBox>
  );
}

TextInput.propTypes = {
  value: propTypes.string,
  onChange: propTypes.func,
  label: propTypes.string.isRequired,
  id:propTypes.string,
  placeholder: propTypes.string
};

export default TextInput;
