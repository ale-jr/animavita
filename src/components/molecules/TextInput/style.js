import styled from "styled-components";

export const TextInputBox = styled.div`
  display: inline-flex;
  flex-direction: column;
  border-radius: 500px;
  border: 1px solid ${({ theme }) => theme.colors.primary.light};
  transition: ${({ theme }) => theme.animations.easeInOut};
  background-color: ${({ theme }) => theme.colors.neutral.white};

  padding: ${({ theme }) => `${theme.spacing.xxs} ${theme.spacing.xs}`};
  &:focus-within {
    border-color: ${({ theme }) => theme.colors.primary.dark};
  }
`;

export const TextInput = styled.input`
  &:focus {
    outline: 0;
  }

  background-color: transparent;
  border: none;
  padding: 0;
  font-size: ${({ theme }) => theme.typography.sizes.xs};
`;

export const TextInputLabel = styled.label`
  font-size: ${({ theme }) => theme.typography.sizes.xxs};
  color: ${({ theme }) => theme.colors.text.medium};
`;
