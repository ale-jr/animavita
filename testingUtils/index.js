import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../src/themes/default";

function ThemeProviderWrapper({ children }) {
  return (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  );
}

export const renderWithTheme = (ui, options) =>
  render(ui, {
    wrapper: ThemeProviderWrapper,
    ...options
  });
