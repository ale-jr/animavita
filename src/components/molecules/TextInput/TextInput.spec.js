import TextInput from ".";
import { renderWithTheme } from "../../../../testingUtils";
import { fireEvent, screen } from "@testing-library/react";

describe("TextInput", () => {
  it("Matches snapshot", () => {
    const { container } = renderWithTheme(
      <TextInput label="Hello" value="xpto" id="input" onChange={() => {}} />
    );
    expect(container).toMatchSnapshot();
  });

  it("Renders label and placeholder correctly", () => {
    const label = "Search";
    const placeholder = "Placeholder";
    renderWithTheme(
      <TextInput label={label} onChange={() => {}} placeholder={placeholder} />
    );

    expect(screen.getByLabelText(label)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(placeholder)).toBeInTheDocument();
  });

  it("Fires onChange event", () => {
    const label = "Search";
    const placeholder = "Placeholder";
    const value = "123";

    const onChange = jest.fn();
    renderWithTheme(
      <TextInput
        label={label}
        onChange={e => onChange(e.target.value)}
        placeholder={placeholder}
      />
    );

    fireEvent.change(screen.getByPlaceholderText(placeholder), {
      target: {
        value
      }
    });

    expect(onChange).toBeCalledWith(value);
  });
});
