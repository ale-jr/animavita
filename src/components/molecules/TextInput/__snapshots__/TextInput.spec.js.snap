// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Matches snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL lmEBxJ"
  >
    <label
      class="sc-dkPtRN bFiZYZ"
      for="input"
    >
      Hello
    </label>
    <input
      class="sc-gsDKAQ jmYbEV"
      id="input"
      value="xpto"
    />
  </div>
</div>
`;
